### PR TITLE
docs: create `<FormLayout />` spec

### PR DESCRIPTION
**Pull request by stephenjwatkins**
## 📝 Changes

- Adds `<FormLayout />` spec

### CODE REVIEW

File: documentation/specs/FormLayout.md [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+type FormLayoutHeaderProps = {
+  children: ReactNode;
+};
+
+type FormLayoutTitleProps = {
+  id?: string;
+  as?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+  children: ReactNode;
+};
```
Comment (ralexmatthews) [inline documentation/specs/FormLayout.md:42]: Question:
What is the purpose of making a purposeful title for the form over using the easy-ui text component?

File: documentation/specs/FormLayout.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+
+type FormLayoutHelperTextProps = {
+  children: ReactNode;
+};
```
Comment (ralexmatthews) [inline documentation/specs/FormLayout.md:46]: Question:
Same with the helper text

**Review by ralexmatthews — APPROVED** [review]
Sweet. Just a few small questions 👍

File: documentation/specs/FormLayout.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+
+type FormLayoutSectionProps = {
+  children: ReactNode;
+};
+
+type FormLayoutGridProps = {
+  children: ReactNode;
+  columns: HorizontalGridProps["columns"];
+};
```
Comment (OskiTheCoder) [inline documentation/specs/FormLayout.md:55]: **question** is the intention for the grid to be more or less a lightweight wrapper around `HorizontalGrid`?

File: documentation/specs/FormLayout.md [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+}
+```
+
+---
+
+## Behavior
+
+### Accessibility
+
+- `FormLayout.Section` is rendered as a `fieldset`. `FormLayout.Title` is rendered as a `legend` within a `FormLayout.Section`.
```
Comment (OskiTheCoder) [inline documentation/specs/FormLayout.md:328]: nice

**Review by OskiTheCoder — APPROVED** [review]

File: documentation/specs/FormLayout.md (hunk 3 above)
Comment (stephenjwatkins) [reply documentation/specs/FormLayout.md:55]: yep exactly. it's a lightweight wrapper around `HorizontalGrid` with gap settings configured to the form layout design requirements

File: documentation/specs/FormLayout.md (hunk 1 above)
Comment (stephenjwatkins) [reply documentation/specs/FormLayout.md:42]: the Title and HelperText components wrap `Text` to lock down the specific variants that design wants to use for the Form and Form.Section title and helper text. this prevents potential accidental inconsistency with form text styles across the product

**Pull request merged**